adapter/avro/src/main/java/org/apache/arrow/adapter/avro/AvroToArrowUtils.java

```diff
@@ -277,11 +333,17 @@ private static Consumer createConsumer(
         break;
       case BYTES:
         if (logicalType instanceof LogicalTypes.Decimal) {
-          arrowType = createDecimalArrowType((LogicalTypes.Decimal) logicalType);
+          LogicalTypes.Decimal decimalType = (LogicalTypes.Decimal) logicalType;
+          arrowType = createDecimalArrowType(decimalType, schema);
           fieldType =
               new FieldType(nullable, arrowType, /* dictionary= */ null, getMetaData(schema));
           vector = createVector(consumerVector, fieldType, name, allocator);
-          consumer = new AvroDecimalConsumer.BytesDecimalConsumer((DecimalVector) vector);
+          if (decimalType.getPrecision() <= 38) {
```

Comment:
Hmm, it's technically possible to have a decimal256 with a smaller precision though

Reply:
I guess in that case it would round-trip to the smaller type?

Reply:
For FIXED decimals I am using the fixedSize to choose the decimal type. For BYTES decimals yes they would just come back as the smaller type if the precision fits.

I used FIXED as the default output for decimals in the producers, because it is closer to the Arrow representation, but on reflection Avro as a format is very focused on keeping data compact, maybe BYTES makes more sense. It seems to be the default choice in Avro. Do you think we should go with that?

Reply:
I think FIXED is okay to allow round-trip by default even if it's not technically as compact.